src/code/z_bgcheck.c

```diff
@@ -4191,10 +4191,10 @@ f32 zdWaterBoxMaxZ = -967.0f;
 s32 WaterBox_GetSurface1(PlayState* play, CollisionContext* colCtx, f32 x, f32 z, f32* ySurface,
```

Comment:
Might as well mention the special case in this function in the description, too.

Reply:
idk, the hardcoded stuff is right there taking 90% of  the body

There's so much hardcoded stuff everywhere it seems more damaging than worth it to comment on every single function that has this kind of thing?

Reply:
And in the Doxygen?

Reply:
I'm overall unsure about the purpose of doxygen generated docs, ngl, what would someone be doing that they don't have the code under their eyes?

In particular this hardcoded thing seems more like a gotcha for mapmaking and I don't expect mapmakers to look at this particular function, neither in source nor in doxygen